### PR TITLE
bugfixes: add-computer & nanodump modules

### PR DESCRIPTION
**Pull request by Marshall-Hallenbeck**
- remove exit(1) calls from add-computer as it was "failing" tests, even though there were no programmatic failures
- properly pass in logger to MSSQL execution (not sure how this didn't come up before, does no one use this?)

### CODE REVIEW

**Comment by NeffIsBack** [comment]
`exit(1)` should be replaced with `return` or not? Otherwise it will probably crash if there is the assumption that some variable exists

**Comment by Marshall-Hallenbeck** [comment]
By default python returns (0)

**Comment by Marshall-Hallenbeck** [comment]
@NeffIsBack readded the exit(1)s to the module options as we discussed

File: nxc/modules/add-computer.py [hunk 1]
```diff
@@ -87,8 +87,7 @@ def on_login(self, context, connection):
         # If SAMR fails now try over LDAPS
         if not self.noLDAPRequired:
             self.do_ldaps_add(connection, context)
-        else:
-            sys.exit(1)
+
```
Comment (NeffIsBack) [inline nxc/modules/add-computer.py:90]: Why removing this one?

File: nxc/modules/add-computer.py [hunk 2]
```diff
@@ -178,7 +177,6 @@ def do_samr_add(self, context):
                     samr.hSamrLookupNamesInDomain(dce, domain_handle, [self.__computerName])
                     self.noLDAPRequired = True
                     context.log.highlight("{}".format('Computer account already exists with the name: "' + self.__computerName + '"'))
-                    sys.exit(1)
```
Comment (NeffIsBack) [inline nxc/modules/add-computer.py:181]: This one should exit as well imo. Otherwise it's still trying to add the computer

Comment (NeffIsBack) [reply nxc/modules/add-computer.py:181]: when you are already on it you could change the line above to a proper format string as well :D

Comment (Marshall-Hallenbeck) [reply nxc/modules/add-computer.py:181]: I don't think we should exit(1) for every single error. The program ran fine, but the computer account already exists.

File: nxc/modules/add-computer.py (hunk 1 above)
Comment (Marshall-Hallenbeck) [reply nxc/modules/add-computer.py:90]: I don't see the reason for it.

File: nxc/modules/add-computer.py (hunk 2 above)
Comment (NeffIsBack) [reply nxc/modules/add-computer.py:181]: Sounds good if you tested it

**Review by NeffIsBack — APPROVED** [review]
LGTM

**Pull request merged**